src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/AvoidUninstantiatedInternalClassesTests.cs

```diff
@@ -223,80 +227,116 @@ Inherits MyAttribute
         }
 
         [Fact]
-        public void CA1812_CSharp_NoDiagnostic_TypeContainingAssemblyEntryPointReturningVoid()
+        public async Task CA1812_CSharp_NoDiagnostic_TypeContainingAssemblyEntryPointReturningVoid()
         {
-            VerifyCSharp(
+            await new VerifyCS.Test
+            {
+                TestCode =
 @"internal class C
 {
     private static void Main() {}
 }",
-            compilationOptions: s_CSharpDefaultOptions.WithOutputKind(CodeAnalysis.OutputKind.ConsoleApplication));
+                SolutionTransforms =
+                {
+                    (solution, projectId) => solution.WithProjectCompilationOptions(projectId, new CSharpCompilationOptions(OutputKind.ConsoleApplication))
```

Comment:
💡 Typically these would transform the existing project options rather than replacing them altogether. It's a bit longer, but supports more scenarios.

```
(solution, projectId) =>
{
  var compilationOptions = solution.GetProject(projectId).CompilationOptions;
  return solution.WithProjectCompilationOptions(projectId, compilationOptions.WithOutputKind(OutputKind.Console));
}
```